service/lambda/api_op_AddLayerVersionPermission.go

```diff
@@ -208,6 +209,52 @@ func (m *opAddLayerVersionPermissionResolveEndpointMiddleware) HandleSerialize(c
 		)
 	}
 
+	authSchemes, err := internalauth.GetAuthenticationSchemes(&resolvedEndpoint.Properties)
+	if err != nil || len(authSchemes) == 0 {
+		return out, metadata, fmt.Errorf("Failed to resolve authentication scheme")
+	}
+
+	supportedAuthSchemeFound := false
+	for _, authScheme := range authSchemes {
+		name := authScheme.GetName()
+		_, supportedAuthSchemeFound = internalauth.SupportedSchemes[name]
+		if supportedAuthSchemeFound {
+			if name == internalauth.SigV4 {
+				v4Scheme, _ := authScheme.(*internalauth.AuthenticationSchemeV4)
+				var signingName, signingRegion string
+				if v4Scheme.SigningName == nil {
+					signingName = "lambda"
+				}
+				if v4Scheme.SigningRegion == nil {
+					signingRegion = m.BuiltInResolver.(*BuiltInResolver).Region
+				}
+				ctx = awsmiddleware.SetSigningName(ctx, signingName)
+				ctx = awsmiddleware.SetSigningRegion(ctx, signingRegion)
+				break
+			}
+			if name == internalauth.SigV4A {
+				v4aScheme, _ := authScheme.(*internalauth.AuthenticationSchemeV4A)
+				var signingName string
+				if v4aScheme.SigningName == nil {
+					signingName = "lambda"
+				}
+				ctx = awsmiddleware.SetSigningName(ctx, signingName)
+				ctx = awsmiddleware.SetSigningRegion(ctx, v4aScheme.SigningRegionSet[0])
+				break
+			}
+			if name == internalauth.None {
+				break
+			}
+		}
+	}
+	if !supportedAuthSchemeFound {
```

Comment:
**fix**: Again the endpoints aren't required to have an auth scheme, if one is found it basically can help influence signing. Otherwise it should just do what it normally would do.